torch/_functorch/aot_autograd.py

```diff
@@ -435,6 +448,47 @@
 FakifiedFlatArgs = NewType("FakifiedFlatArgs", List[Any])
 
 
+TOutputCode = TypeVar("TOutputCode", bound=OutputCode)
+
+
+class AOTDispatchCompiler(Protocol):
+    """
+    Represents a fw or bw_compiler passed to AOTAutograd.
+    """
+
+    def __call__(
+        self,
+        gm: torch.fx.GraphModule,
+        example_inputs: Sequence[InputType],
+    ) -> Any:
+        ...
+
+
+# TODO: bikeshed on this name
+class AOTDispatchCompilerWithOutput(AOTDispatchCompiler):
```

Comment:
should "cacheable" show up somewhere in the name, if the main invariant we're requiring is that backends using this class must return a cache-friendly output code?

Reply:
Instead of it directly referencing cache, I wanted it to be a bit more general because serializability > cacheability. Maybe `SerializableAOTDispatchCompiler`? 

But that makes it sound like the compiler itself is serializable...

Reply:
I guess the return type implies it too